internal/providers/exoscale/exoscale.go

```diff
@@ -37,7 +37,9 @@ var (
 
 // FetchConfig fetch Exoscale ign user-data config
 func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
-	data, err := f.FetchToBuffer(userdataURL, resource.FetchOptions{})
+	data, err := f.FetchToBuffer(userdataURL, resource.FetchOptions{
```

Comment:
This assumes the metadata service always compresses, and will regress in any case where compression doesn't happen.  Do we have an authoritative reference saying that compression is always enabled?  What does cloud-init do?

Reply:
Requested more info from Pierre on this.

Reply:
I'll have a look internally to reply to your question

Reply:
Then I come back to you with an unpleasant reply.

`user-data` is uploaded as it is in the server, so it can be only **plain text** or **Gzip** from our tooling/portal perspective. Since you were using our CLI to test the images with Exoscale Instances, CLI is Gzipping the user-data you give.

The not very pretty patch I propose, is to try to check if it's Gzip by reading the header, then fallback to plain text on failure. Keep the same implementation for 0 sized data or `404` replies.

Reply:
Okay, good to know, and thanks for checking.  Ignition tries very hard to avoid format sniffing.  Since Exoscale is just serving the provided userdata as-is, I assume there's nothing in HTTP response headers to indicate that the data is compressed?  Is it possible to tell the CLI not to compress the userdata before uploading?

Reply:
If you try to register a template, from the web portal, it should be not compressed.

I think a CLI change has been made in the past, I'll do a PR in the CLI to choose such an option.
I would say, by default, not compressing the file is better design.

> I assume there's nothing in HTTP response headers to indicate that the data is compressed?
Furthermore, I need to check with curl on an Exoscale Instance.

Reply:
And in egoscale package it did not compress by default:
v1: https://github.com/exoscale/egoscale/blob/master/virtual_machines.go#L287
v2: https://github.com/exoscale/egoscale/blob/master/v2/instance.go#L39 (generated code from api v2)
it's only base64 encoded

So, I'll make PR in the CLI

Reply:
Does this mean you're going to change the CLI to not compress the userdata so it will have the same behavior as the web UI?

Reply:
I can make a flag to compress or not compress the user-data.
IMHO, better to not compressed by default, we will see at the review

Reply:
What would be nice is consistency. Ideally either have the CLI **and** Web UI always compress, or have them both **never** compress.

Reply:
Here is the PR in the Exoscale CLI https://github.com/exoscale/cli/pull/451

Reply:
Yeah, I think that's the best approach.  We can document the minimum required version of the CLI, and that an explicit `--cloud-init-compress` is not supported.